### PR TITLE
Update structure and use webpack

### PR DESCRIPTION
**Pull request by KABBOUCHI**
- Update structure (more friendly)
- Add config file
- Add webpack: client will use the precompiled version
- Auth

user no longer need to add manually, install or add options

image

N.B: Not fully tested, please check if Controller need updates 

Inspired by Laravel Horizon

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,32 +28,6 @@ A file based translation manager, which unlike other Lang managers don't need a
...
 
-- after installation, package will auto-add
-    + package routes to `routes/web.php`
-    + package assets compiling to `webpack.mix.js`
-    + `MIX_LINGO_FRAMEWORK=bulma` to `.env`
-
-- install dependencies
-
-```bash
-yarn add vue axios vue-ls vuemit vue-notif vue-clipboard2 vue-tippy vue-awesome
```
Comment (ctf0) [inline README.md:39]: the reason for keeping the dependencies external is so you can use the same package anywhere else without the need to reinstall it

File: README.md [hunk 2]
```diff
@@ -63,22 +37,4 @@ new Vue({
...
 - support all laravel different translation key formats.
-- directly copy translation key by clicking on the tool-tip.
-
-<br>
-
-## Usage
-
-- for styling we use ***bulma***
-
-> ***Or Use another Framework***
```
Comment (ctf0) [inline README.md:74]: not sure why ppl get confused with this, but u r not the first one to fall into this quote trap.

File: config/lingo.php [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+return [
+    'uri'        => 'lingo',
+    'as'         => 'lingo.',
+    'middleware' => 'web',
+    'theme'      => env('LINGO_THEME','bulma'),
+];
```
Comment (ctf0) [inline config/lingo.php]: why do u need this ?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    "vuemit": "^1.0.7"
+  },
+  "devDependencies": {
+    "babel-preset-env": "^1.6.1",
+    "babel-preset-stage-3": "^6.24.1",
+    "cross-env": "^5.0.1",
+    "es6-promise": "^4.0.5",
+    "laravel-mix": "^1.4.2"
+  }
+}
```
Comment (ctf0) [inline package.json:29]: again why do u need this ? other than restricting the packages versions

File: public/css/app.css [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
...tcbiAgICB6LWluZGV4OiAxO1xuICAgIHRvcDogMzBweDtcbiAgICBiYWNrZ3JvdW5kLWNvbG9yOiB3aGl0ZTtcbn1cblxuXG5cbi8vIFdFQlBBQ0sgRk9PVEVSIC8vXG4vLyAuL3Jlc291cmNlcy9hc3NldHMvc2Fzcy9yZXNvdXJjZXMvYXNzZXRzL3Nhc3MvYXBwLnNjc3MiLCJ0ZCxcbnRoIHtcbiAgdGV4dC1hbGlnbjogY2VudGVyICFpbXBvcnRhbnQ7XG4gIHZlcnRpY2FsLWFsaWduOiBtaWRkbGUgIWltcG9ydGFudDtcbn1cblxuLm0tdC01MCB7XG4gIG1hcmdpbi10b3A6IDUwcHg7XG59XG5cbi5tLXQtMTAge1xuICBtYXJnaW4tdG9wOiAxMHB4O1xufVxuXG4uc2xpZGUtZmFkZS1lbnRlci1hY3RpdmUsXG4uc2xpZGUtZmFkZS1sZWF2ZS1hY3RpdmUge1xuICB0cmFuc2l0aW9uOiBhbGwgMC40cztcbn1cblxuLnNsaWRlLWZhZGUtZW50ZXIsXG4uc2xpZGUtZmFkZS1sZWF2ZS10byB7XG4gIHRyYW5zZm9ybTogdHJhbnNsYXRlWCgxMHB4KTtcbiAgb3BhY2l0eTogMDtcbn1cblxuLmNvbXAtZmFkZS1lbnRlci1hY3RpdmUsXG4uY29tcC1mYWRlLWxlYXZlLWFjdGl2ZSB7XG4gIHRyYW5zaXRpb246IGFsbCAwLjRzO1xufVxuXG4uY29tcC1mYWRlLWVudGVyLFxuLmNvbXAtZmFkZS1sZWF2ZS10byB7XG4gIHRyYW5zZm9ybTogdHJhbnNsYXRlWCgtMzBweCk7XG4gIG9wYWNpdHk6IDA7XG59XG5cbi5uZXN0ZWRLZXkge1xuICBjb2xvcjogd2hpdGU7XG4gIGJhY2tncm91bmQtY29sb3I6ICMzNjM2MzY7XG59XG5cbnRoZWFkIHRoIHtcbiAgcG9zaXRpb246IHN0aWNreTtcbiAgei1pbmRleDogMTtcbiAgdG9wOiAzMHB4O1xuICBiYWNrZ3JvdW5kLWNvbG9yOiB3aGl0ZTtcbn1cblxuXG5cblxuLy8gV0VCUEFDSyBGT09URVIgLy9cbi8vIC4vYXBwLnNjc3MiXSwic291cmNlUm9vdCI6IiJ9*/
```
Comment (ctf0) [inline public/css/app.css]: what about the `bulma` override styles ?

File: README.md (hunk 1 above)
Comment (KABBOUCHI) [reply README.md:39]: No need to install any dependencies, only the dev need them

File: README.md (hunk 2 above)
Comment (KABBOUCHI) [reply README.md:74]: You can use other style:
1- .env (or config('lingo.theme')) => LINGO_THEME=theme1
2- publish lingo-views and add new file lingo-theme1.blade.php


image

File: config/lingo.php (hunk 3 above)
Comment (KABBOUCHI) [reply config/lingo.php]: add ability for user to change
- the prefix (route)
- the theme from config file or .env (previously was MIX_LINGO_FRAMEWORK) 
- custom middleware

File: package.json (hunk 4 above)
Comment (KABBOUCHI) [reply package.json:29]: You always need to restrict package version,  from v1 to v2 can break things

and again this for the developer

File: public/css/app.css (hunk 5 above)
Comment (KABBOUCHI) [reply public/css/app.css]: default will be bulma.css, you can support other framework like bootstrap

when user need to add custom style he can create new theme and change the file

image

**Comment by KABBOUCHI** [comment]
I added some new updates like Lingo::Auth

**Comment by ctf0** [comment]
first thanx so much for all ur help

- the config is not really needed, because for the routes in its current form, you are free to do whatever you want with it, you can put it under any group of middlewares, prefixes, etc..

config should be used only when its needed, specially as atm laravel config folder is a big mess.

- for the webpack, could you tell how will i use it in a fresh laravel install ?

- for changing the folder structure, i prefer to keep all the package files under the same level `src`, so plz undo that.

- all my packages are made toward devs, so maintaining a compiled `css, js` files is simply a big headache for me.

- for the controller, am not sure what exactly have you changed other than using the config values.

- am not a big fan of the horizon auth approach, so this also will be declined.

**Comment by ctf0** [comment]
@KABBOUCHI why closed ?

**Comment by KABBOUCHI** [comment]
you declined all my updates, why you need it :)

edit:

all my updates does is for the end user benefits, if you don't want to maintain your product, it's okay no need to merge it

**Comment by ctf0** [comment]
how about this 
> for the webpack, could you tell how will i use it in a fresh laravel install ?

**Comment by KABBOUCHI** [comment]
Not sure what do you want, I added webpack to precompile css/js for dev purpose

it's already there in a fresh laravel installer

**Comment by ctf0** [comment]
okay, thanx again

**Pull request closed without merging**